From e2ba48ba68fc271ce9f8052c4d66230bef99e6e5 Mon Sep 17 00:00:00 2001
From: Matthew Wong <mattwon@amazon.com>
Date: Thu, 26 Aug 2021 14:41:37 -0700
Subject: [PATCH] --EKS-PATCH-- Fix TestAddFlags --proxy-cidr-whitelist case

--proxy-cidr-whitelist/--proxy-cidr-allowlist is a flag added by EKS
---
 cmd/kube-apiserver/app/options/options_test.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmd/kube-apiserver/app/options/options_test.go b/cmd/kube-apiserver/app/options/options_test.go
index 3a833ff188f..72ecc46c982 100644
--- a/cmd/kube-apiserver/app/options/options_test.go
+++ b/cmd/kube-apiserver/app/options/options_test.go
@@ -111,6 +111,7 @@ func TestAddFlags(t *testing.T) {
 		"--kubelet-client-certificate=/var/run/kubernetes/ceserver.crt",
 		"--kubelet-client-key=/var/run/kubernetes/server.key",
 		"--kubelet-certificate-authority=/var/run/kubernetes/caserver.crt",
+		"--proxy-cidr-whitelist=192.168.0.0/24",
 		"--proxy-client-cert-file=/var/run/kubernetes/proxy.crt",
 		"--proxy-client-key-file=/var/run/kubernetes/proxy.key",
 		"--request-timeout=2m",
@@ -127,7 +128,7 @@ func TestAddFlags(t *testing.T) {
 		MasterCount:            5,
 		EndpointReconcilerType: string(reconcilers.LeaseEndpointReconcilerType),
 		AllowPrivileged:        false,
-		ProxyCIDRAllowlist:     kubeoptions.NewIPNetSlice("0.0.0.0/0"),
+		ProxyCIDRAllowlist:     kubeoptions.NewIPNetSlice("192.168.0.0/24"),
 		GenericServerRunOptions: &apiserveroptions.ServerRunOptions{
 			AdvertiseAddress:            net.ParseIP("192.168.10.10"),
 			CorsAllowedOriginList:       []string{"10.10.10.100", "10.10.10.200"},
